app/database/itemStore.go

```diff
@@ -119,3 +141,42 @@ func checkAccessForID(id int64, last bool, accDets []itemAccessDetails) error {
 	// no row matching this item_id
 	return fmt.Errorf("not visible item_id %d", id)
 }
+
+func (s *ItemStore) checkIfItemIsRoot(id int64) (bool, error) {
+	count := 0
+	if err := s.ByID(id).Count(&count).Error(); err != nil {
+		return false, err
+	}
+	if count == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (s *ItemStore) checkHierarchicalChain(ids []int64) (bool, error) {
+	if len(ids) == 1 {
+		return true, nil
+	}
+
+	db := s.ItemItems().All()
+	previousID := ids[0]
+	for index, id := range ids {
+		if index == 0 {
+			continue
+		}
+
+		db = db.Or("idItemParent=? AND idItemChild=? AND iChildOrder=1", previousID, id)
```

Comment:
Dor `iChildOrder`, it is used to set the order of the child among all the parents children. (it is not the depth). Even if it is not the first child, it is still a child.